### PR TITLE
Dropdown Component Start

### PR DESCRIPTION
**Pull request by andrew-frueh**
The beginnings of a dropdown component based off of the html select tag.  This dropdown is used for choosing from a list of selectable items.  There is also a dropdown option in our button component, but that is used for a list of action items (e.g. "Delete all selected").  Look forward to your feedback.

dropdown

### CODE REVIEW

File: src/app/lib/dropdown/dropdown.component.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'hc-dropdown',
+  templateUrl: './dropdown.component.html',
+  styleUrls: ['./dropdown.component.scss']
+})
+export class DropdownComponent implements OnInit {
```
Comment (corytak) [inline src/app/lib/dropdown/dropdown.component.ts]: This is an angular deep track, but I believe we'll want to integrate this dropdown with reactive forms by implementing the `ControlValueAccessor` interface. 
[Custom Form Controls](https://alligator.io/angular/custom-form-control/)

File: src/app/lib/dropdown/dropdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'hc-dropdown',
```
[the remark this answers is not in the document]
Comment (andrew-frueh) [reply src/app/lib/dropdown/dropdown.component.ts]: Blech - I actually thought about that @corykon when I was building it, and somehow convinced myself that Dropdown was a more descriptive name.  But now that I'm giving it another look, it does seem that Select is the more accepted way to go.  Well hopefully some careful find/replace work will keep this from being too painful.

So just to be clear before I do that, we're proposing that I call it "SelectComponent", correct? @benjanderson @corytak

File: src/app/lib/dropdown/dropdown.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  templateUrl: './dropdown.component.html',
+  styleUrls: ['./dropdown.component.scss']
+})
+export class DropdownComponent implements OnInit {
+
```
[the remark this answers is not in the document]
Comment (andrew-frueh) [reply src/app/lib/dropdown/dropdown.component.ts]: Good call - I forgot to include anything for that.  I'll put that on my todo list for this one.

File: src/app/lib/dropdown/dropdown.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  @Input() placeholder: string;
+  @Input() options = [];
+  @Input() width: string;
+
+  selectWidth = '350px';
```
[the remark this answers is not in the document]
Comment (andrew-frueh) [reply src/app/lib/dropdown/dropdown.component.ts]: Interesting, I hadn't thought about that - but I could see how that would help from a responsive design perspective.  And judging by Ben and Cory Tak's thumbs up, that looks to be to good with others too - so I'll adjust as you suggested.

**Comment by andrew-frueh** [comment]
@corykon as per your suggestions, I adjusted the width to 100% and wrapped it in a fixed width div in the demo page, I changed the name of the component to "Select", and I added a disabled flag that can be set.  Can you take a look as soon as you get a chance and let me know if this is good to check in?  Then I'll add an issue @corytak and @benjanderson asking for a developer's help to do a v1.1 version that adds the necessary form code to make this component more useful.

**Pull request merged**